### PR TITLE
darray write with async for serial iotypes

### PR DESCRIPTION
**Pull request by edhartnett**
In this PR the first test case of writing with async and darrays is added to the code base. Only serial iotypes are supported in this PR (PIO_IOTYPE_NETCDF, PIO_IOTYPE_NETCDF4C). Support for parallel IO types is coming in a future PR.

There is still work to be done here on the test, but let's get this progress merged and I will continue to develop testing in future PRs.

Part of #89.
Part of #388.

The new test test_darray_async is only run when PIO_USE_MALLOC=On. (The bget code does not work with async).

I have merged to develop for testing.

### CODE REVIEW

File: src/clib/pioc_support.c [hunk 1]
```diff
@@ -1718,7 +1727,7 @@ int PIOc_createfile_int(int iosysid, int *ncidp, int *iotype, const char *filena
     for (int i = 0; i < PIO_MAX_VARS; i++)
     {
         file->varlist[i].record = -1;
-        file->varlist[i].ndims = -1;
+        /* file->varlist[i].ndims = -1; */
```
Comment (jedwards4b) [inline src/clib/pioc_support.c]: Should we just remove this commented code?

File: tests/cunit/CMakeLists.txt [hunk 2]
```diff
@@ -74,13 +74,17 @@ if (NOT PIO_USE_MPISERIAL)
...
   add_executable (test_darray_3d EXCLUDE_FROM_ALL test_darray_3d.c test_common.c)
-  target_link_libraries (test_darray_3d pioc)  
+  target_link_libraries (test_darray_3d pioc)
   add_executable (test_decomp_uneven EXCLUDE_FROM_ALL test_decomp_uneven.c test_common.c)
   target_link_libraries (test_decomp_uneven pioc)  
   add_executable (test_decomps EXCLUDE_FROM_ALL test_decomps.c test_common.c)
   target_link_libraries (test_decomps pioc)
   add_executable (test_rearr EXCLUDE_FROM_ALL test_rearr.c test_common.c)
   target_link_libraries (test_rearr pioc)
+  if (PIO_USE_MALLOC)
```
Comment (jedwards4b) [inline tests/cunit/CMakeLists.txt]: Do these tests only work with MALLOC? Why?

File: src/clib/pioc_support.c (hunk 1 above)
Comment (edhartnett) [inline src/clib/pioc_support.c]: Yes, I will remove it.

File: tests/cunit/CMakeLists.txt (hunk 2 above)
Comment (edhartnett) [inline tests/cunit/CMakeLists.txt]: Yes, something about bget breaks for async. Probably it is not initializing buffers correctly.

**Pull request merged**